### PR TITLE
Add a terminal based Oauth flow to avoid selenium

### PR DESCRIPTION
**Pull request by pavanagrawal123**
Selenium can be pretty annoying to install and I personally find it frustrating to work with. I implemented an Oauth flow here that can be completed over the terminal, so we can save other people the pain of Selenium!

edit: I also noticed that nose was missing in the requirements.txt, so I added it in. Should we go ahead and lock its version? 

Also when I run `make fix` locally, I'm getting a bunch of new formatting changes, have formatting changes not been run on the repo? Maybe some amount of formatting config is missing locally for me.

### CODE REVIEW

**Comment by nicmcd** [comment]
I personally would love this feature as I'm running my TDA app on a headless server.

File: tda/auth.py [hunk 1]
```diff
@@ -163,10 +164,79 @@ def client_from_login_flow(webdriver, api_key, redirect_url, token_path,
...
+    print('Open the following page in a browser. Login, and then paste the URL that your browser is sent to below.')
+    print()
+    print(authorization_url)
+    print()
+    print('If you encounter any issues, see here for troubleshooting: ' +
+          'https://tda-api.readthedocs.io/en/stable/auth.html' +
+          '#troubleshooting')
+    print('\n**************************************************************\n')
+
+    current_url = input("Redirected URL:  ")
```
Comment (h00shi) [inline tda/auth.py:218]: I tested your function on Mac and it seems not to work. For some reason python's input() hangs for large inputs. Instead trying the prompt function from `prompt_toolkit` works flawlessly.

Comment (pavanagrawal123) [reply tda/auth.py:218]: That's quite weird. For me, on MacOS as well, I experienced the same hang on Python 3.8, but was fixed after importing the "readline" module. Let me test again with 3.9 on MacOS and figure out if we need to pull in the external module.

**Comment by wolframarnold** [comment]
This branch worked for me, from a Colab notebook. I'd love to see it merged.

**Comment by alexgolec** [comment]
Thanks for submitting this PR, but it seems it's rotted a little. Apologies for the delay. I've written up my own implementation here: 

https://github.com/alexgolec/tda-api/pull/139

**Pull request closed without merging**